builtin/credential/cert/path_login.go

```diff
@@ -170,49 +186,56 @@ func (b *backend) verifyCredentials(req *logical.Request) (*ParsedCert, *logical
 		return nil, logical.ErrorResponse("invalid certificate or no client certificate supplied"), nil
 	}
 
-	validChain := b.checkForValidChain(trustedChains)
-	if !validChain {
-		return nil, logical.ErrorResponse(
-			"no chain containing non-revoked certificates could be found for this login certificate",
-		), nil
+	// Search for a ParsedCert that intersects with the validated chains and any additional constraints
+	matches := make([]*ParsedCert, 0)
+	for _, trust := range trusted { // For each ParsedCert in the config
+		for _, tCert := range trust.Certificates { // For each certificate in the entry
+			for _, chain := range trustedChains { // For each root chain that we matched
+				for _, cCert := range chain { // For each cert in the matched chain
+					if tCert.Equal(cCert) && // ParsedCert intersects with matched chain
+						b.matchesConstraints(connState.PeerCertificates[0], chain, trust) { // validate client cert + matched chain against the config
```

Comment:
s/connState.PeerCertificates[0]/clientCert. This would need increasing the scope of clientCert above.